src/CodeGen_Hexagon.cpp

```diff
@@ -2324,11 +2183,9 @@ bool CodeGen_Hexagon::possiblyCodeGenWideningMultiply(const Mul *op) {
     Ops.clear();
     //   Vdd.uh=vmpy(Vu.ub,Rt.ub)
     //   Vdd.uw=vmpy(Vu.uh,Rt.uh)
-    Patterns.push_back(Pattern(WPICK(wild_u16x128, wild_u16x64) *
-                               WPICK(wild_u16x128, wild_u16x64),
+    Patterns.push_back(Pattern(wild_u16x2W * wild_u16x2W,
```

Comment:
I'd expect the result to be the widened element type though. Basically, I'd expect this Func/schedule to hit this pattern:

    Func f("f");
    Expr a = u16(...);
    Expr b = u16(...);
    f(x) = u32(a) * u32(b);
    f.vectorize(x, 32);   // On HVX_64

However, you'd actually have to vectorize it twice as wide as the width of a vector of a/b to hit that pattern (and you get a 4W vector as the result).